frontend/src/components/common/Sidebar.tsx

```diff
@@ -0,0 +1,233 @@
+          color: "black",
+        }}
+      >
+        <Box p={1}>
+          <BookIcon />
+        </Box>
+        <Box>{navText}</Box>
+        <Spacer />
+        <Box
+          onClick={(e) => {
+            e.preventDefault();
+            setShowMore(!showMore);
+          }}
+        >
+          {existSubPage && (showMore ? <ChevronUpIcon /> : <ChevronDownIcon />)}
+        </Box>
+      </Flex>
+    </Link>
+  );
+};
+
+const UpperNavItem = ({ navText, urlLink, subPages }: UpperNavItemProps) => {
+  const [showMore, setShowMore] = useState(false);
+  let existSubPage;
+  let displaySub;
+  if (subPages) {
+    existSubPage = true;
+    displaySub = subPages.map((page) => (
+      <LowerNavItem key={page.name} name={page.name} urlLink={page.urlLink} />
+    ));
+  } else {
+    displaySub = null;
+    existSubPage = false;
+  }
+
+  return (
+    <>
+      <NavItem
+        showMore={showMore}
+        setShowMore={setShowMore}
+        key={navText}
+        navText={navText}
+        urlLink={urlLink}
+        existSubPage={existSubPage}
+      >
+        {navText}
+      </NavItem>
+      {showMore && displaySub}
+    </>
+  );
+};
+
+interface MobileProps extends FlexProps {
+  onOpen: () => void;
+}
+const MobileNav = ({ onOpen, ...rest }: MobileProps) => {
+  return (
+    <Flex
+      ml={{ base: 0, md: 60 }}
+      px={{ base: 4, md: 24 }}
+      height="20"
+      alignItems="center"
+      bg={useColorModeValue("white", "gray.900")}
+      borderBottomWidth="1px"
+      borderBottomColor={useColorModeValue("gray.200", "gray.700")}
+      justifyContent="flex-start"
+      {...rest}
+    >
+      <IconButton variant="outline" onClick={onOpen} aria-label="open menu" />
+
+      <Text fontSize="2xl" ml="8" fontFamily="monospace" fontWeight="bold">
+        Logo
+      </Text>
+    </Flex>
+  );
+};
+
+const SidebarContent = ({ onClose, linkItems, ...rest }: SidebarProps) => {
+  return (
+    <Box
+      bg={useColorModeValue("white", "gray.900")}
+      borderRight="1px"
+      borderRightColor={useColorModeValue("gray.200", "gray.700")}
+      w={{ base: "full", md: 60 }}
+      pos="fixed"
+      h="full"
+      {...rest}
+    >
+      <Flex h="20" alignItems="center" mx="8" justifyContent="space-between">
+        <Text fontSize="2xl" fontFamily="monospace" fontWeight="bold">
+          Logo
+        </Text>
+        <CloseButton display={{ base: "flex", md: "none" }} onClick={onClose} />
+      </Flex>
+      {linkItems.map((link) => (
+        <UpperNavItem
+          key={link.name}
+          navText={link.name}
+          urlLink={link.urlLink}
+          subPages={link.subPages}
+        >
+          {link.name}
+          {link.subPages}
+        </UpperNavItem>
+      ))}
+    </Box>
+  );
+};
+const Sidebar = ({ linkItems }: LinkItemArrayProps): React.ReactElement => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <Box minH="100vh" bg={useColorModeValue("gray.100", "gray.900")}>
```

Comment:
just wondering if `useColorModeValue` is necessary? (are we supporting different color modes?)

Reply:
just wondering if `useColorModeValue` is necessary? (are we supporting different color modes?)